Remove sec key

ayinloya opened the pull request:
This change uses signature as the default for signing requests

To test, pull the changes and send a verification job

ayinloya committed:
Remove sec key from some classes

ayinloya committed:
Fix failing test

ayinloya committed:
Update Readmin: Remove sec_ke references
Bumb version to 1.1.0

ayinloya committed:
Fix lint issues

ayinloya committed:
Rerun lint

ayinloya committed:
Make signature default

ayinloya committed:
run lint

mldangelo commented on pyproject.toml:
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "smile-id-core"
-version = "1.0.9"
+version = "1.1.0"
```
this needs to be a 2.0.0 because of the breaking API change, but don't change it until we issue a release. 
```suggestion
version = "1.0.9"
```

mldangelo reviewed (changes requested):
Hey @ayinloya, can you control + f this repo for `sec_key` and remove the other references?

ayinloya commented:
> Hey @ayinloya, can you control + f this repo for `sec_key` and remove the other references?

Noted, will check it

ayinloya committed:
Remove references to sec_key

ayinloya committed:
Revert to version 1.0.9

ayinloya committed:
Fix type error

vanshg commented on README.md:
````diff
@@ -477,13 +479,13 @@ from smile_id_core import Signature
 ```json
 {
-    "sec_key": "<the generated sec key>",
+    "signature": "<the generated sec key>",
````
```suggestion
    "signature": "<the generated signature>",

```

vanshg commented on smile_id_core/Signature.py:
```diff
@@ -17,42 +17,16 @@ def __init__(self, partner_id: str, api_key: str):
-        _timestamp = timestamp
-        if _timestamp is None:
-            _timestamp = datetime.now().isoformat()
+    def generate_signature(self, timestamp=datetime.now().isoformat()):
```
Unfortunately, python caches the default parameter results, so this would be the same value every time the function is called. Therefore, we should default this to `None`, and then assign a default value within the function code itself.

i.e.
```python
>>> def test(t=datetime.now().isoformat()):
...     print(t)
... 
>>> test()
2022-11-14T08:47:10.375300
>>> test()
2022-11-14T08:47:10.375300
>>> test()
2022-11-14T08:47:10.375300
>>> test()
2022-11-14T08:47:10.375300
```

See also: https://docs.python-guide.org/writing/gotchas/#what-you-wrote

vanshg commented on smile_id_core/Signature.py:
```diff
@@ -17,42 +17,16 @@ def __init__(self, partner_id: str, api_key: str):
         }
 
-    def confirm_signature(self, timestamp, msg_signature):
+    def confirm_signature(self, timestamp: str, msg_signature):
```
Can you complete the type signature for the entire function? i.e. provide a type for `msg_signature` and for the return value of the function

vanshg replied:
@ayinloya This one seems unresolved to me

I think it should be something like this

```suggestion
    def confirm_signature(self, timestamp: str, msg_signature: str) -> bool:
```

vanshg commented on smile_id_core/WebApi.py:
```diff
@@ -116,7 +116,7 @@ def submit_job(
                 id_info_params=id_info_params,
                 upload_url=upload_url,
-                sec_params=sec_params,
+                signature_params=sec_params,
```
```suggestion
                signature_params=signature_params,
```

I'd imagine the declaration of `sec_params` also needs to be updated

vanshg commented on smile_id_core/WebApi.py:
```diff
@@ -196,14 +194,10 @@ def __validate_return_data(self, options: Dict) -> None:
         self, partner_params: Dict, sec_params: Dict, use_enrolled_image: bool
     ) -> Dict:
-        validate_sec_params(sec_params)
+        validate_signature_params(sec_params)
```
`sec_params` -> `signature_params`

ayinloya committed:
Remove sec key reference in readme

ayinloya committed:
Fix generate_signature method

ayinloya committed:
Remove _get_security_key_params method

ayinloya committed:
Fixed Missing argument "api_key"

mldangelo commented on smile_id_core/Utilities.py:
```diff
@@ -31,14 +31,12 @@ def get_job_status(
         partner_params: Dict,
         option_params: Dict,
-        sec_params: Optional[Dict] = None,
+        signature: Optional[Dict] = None,
```
This is a breaking API change. I think we should make it, we just need to talk about it in the changelog.

mldangelo commented on smile_id_core/WebApi.py:
```diff
@@ -196,14 +191,10 @@ def __validate_return_data(self, options: Dict) -> None:
-        return sec_key_gen.generate_sec_key()
-
     def __prepare_prep_upload_payload(
         self, partner_params: Dict, sec_params: Dict, use_enrolled_image: bool
```
```suggestion
        self, partner_params: Dict, signature_params: Dict, use_enrolled_image: bool
```

mldangelo commented on smile_id_core/WebApi.py:
```diff
@@ -220,12 +211,12 @@ def poll_job_status(
         partner_params: Dict,
         options_params: Dict,
-        sec_params: Optional[Dict],
+        signature_params: Optional[Dict],
```
this is another breaking api change that needs to go into the changelog.

mldangelo commented on tests/test_utilities.py:
```diff
@@ -21,7 +21,7 @@ def setUp(self):
         self.cipher = PKCS1_v1_5.new(self.key.exportKey())
         self.__reset_params()
-        self.utilities = Utilities(self.partner_id, self.public_key, 0)
+        self.utilities = Utilities(self.partner_id, self.api_key, 0)
```
can you explain this change?

ayinloya replied:
Public key is a of type `byte` but Utilities accepts a `str`, this breaks the test when `api_key.encode("utf-8")` is called when generating a signature.

mldangelo reviewed (changes requested):
This is really good. I left a few short comments. Happy to LGTM after you address them. Thanks so much @ayinloya

ayinloya committed:
Update changelog to reflect change in method params
Change method param names in __prepare_prep_upload_payload, poll_job_status and get_job_status

ayinloya committed:
Update changelog

mldangelo reviewed (approved):
Thank you!

vanshg commented:
LGTM. Should we also be bumping the version number?

mldangelo committed:
Merge branch 'main' into remove-sec-key

mldangelo committed:
fix changelog

mldangelo committed:
lint

mldangelo reviewed (approved):
Thanks @ayinloya
